drivers/wifi/infineon/airoc_wifi.c

```diff
@@ -0,0 +1,652 @@
+	airoc_wifi_data.stats.bytes.received += len;
+	airoc_wifi_data.stats.pkts.rx++;
+#endif
+
+	if (net_pkt_unref_flag) {
+		net_pkt_unref(pkt);
+#if defined(CONFIG_NET_STATISTICS_WIFI)
+		airoc_wifi_data.stats.errors.rx++;
+#endif
+	}
+}
+
+static void *link_events_handler(whd_interface_t ifp, const whd_event_header_t *event_header,
+				 const uint8_t *event_data, void *handler_user_data)
+{
+	ARG_UNUSED(ifp);
+	ARG_UNUSED(event_data);
+	ARG_UNUSED(handler_user_data);
+
+	(void)k_msgq_put(&airoc_wifi_msgq, event_header, K_FOREVER);
+	return NULL;
+}
+
+static void airoc_event_task(void)
+{
+	whd_event_header_t event_header;
+
+	while (1) {
+		(void)k_msgq_get(&airoc_wifi_msgq, &event_header, K_FOREVER);
+
+		switch ((whd_event_num_t)event_header.event_type) {
+		case WLC_E_LINK:
+			break;
+
+		case WLC_E_DEAUTH_IND:
+		case WLC_E_DISASSOC_IND:
+			net_if_dormant_on(airoc_wifi_iface);
+			break;
+
+		default:
+			break;
+		}
+	}
+}
+
+static void airoc_mgmt_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct airoc_wifi_data *data = dev->data;
+	struct ethernet_context *eth_ctx = net_if_l2_data(iface);
+
+	eth_ctx->eth_if_type = L2_ETH_IF_TYPE_WIFI;
+	data->iface = iface;
+	airoc_wifi_iface = iface;
+
+	/* Read WLAN MAC Address */
+	(void)whd_wifi_get_mac_address(airoc_if, &airoc_if->mac_addr);
+	(void)memcpy(&data->mac_addr, &airoc_if->mac_addr, sizeof(airoc_if->mac_addr));
+
+	/* Assign link local address. */
+	(void)net_if_set_link_addr(iface, data->mac_addr, 6, NET_LINK_ETHERNET);
+
+	(void)ethernet_init(iface);
+
+	/* Not currently connected to a network */
+	(void)net_if_dormant_on(iface);
+
+	/* L1 network layer (physical layer) is up */
+	(void)net_if_carrier_on(data->iface);
+}
+
+whd_result_t cy_buffer_pool_init(void *tx_packet_pool, void *rx_packet_pool)
+{
+	CY_UNUSED_PARAMETER(tx_packet_pool);
+	CY_UNUSED_PARAMETER(rx_packet_pool);
+
+	return 0;
+}
+
+static void scan_cb_search(whd_scan_result_t **result_ptr, void *user_data,
+			   whd_scan_status_t status)
+{
+	if (status == WHD_SCAN_ABORTED) {
+		k_sem_give(&airoc_wifi_data.sema_scan);
+		return;
+	}
+	if (status == WHD_SCAN_COMPLETED_SUCCESSFULLY) {
+		k_sem_give(&airoc_wifi_data.sema_scan);
+	} else if (status == WHD_SCAN_INCOMPLETE) {
+		if (user_data != NULL) {
+			if ((**result_ptr).SSID.length > 0) {
+				if (strcmp((**result_ptr).SSID.value,
+					   ((whd_scan_result_t *)user_data)->SSID.value) == 0) {
+					memcpy(user_data, *result_ptr, sizeof(whd_scan_result_t));
+				}
+			}
+		}
+	}
+}
+
+static void parse_scan_result(whd_scan_result_t *p_whd_result, struct wifi_scan_result *p_zy_result)
+{
+	if (p_whd_result->SSID.length != 0) {
+		p_zy_result->ssid_length = p_whd_result->SSID.length;
+		sprintf(p_zy_result->ssid, "%s", p_whd_result->SSID.value);
+		p_zy_result->channel = p_whd_result->channel;
+		p_zy_result->security = (p_whd_result->security == 0) ? WIFI_SECURITY_TYPE_NONE
+								      : WIFI_SECURITY_TYPE_PSK;
```

Comment:
I see WPA3 support in some chips, so, may be an explicit check for WPA2?

Reply:
updated